mopidy_spotify/playback.py

```diff
@@ -187,8 +195,14 @@ def music_delivery_callback(
         bytes(frames), timestamp=buffer_timestamp.get(), duration=duration
     )
 
-    # We must block here to know if the buffer was consumed successfully.
-    consumed = audio_actor.emit_data(buffer_).get()
+    # If there is any held buffer, send it
+    if _held_buffer:
+        consumed = audio_actor.emit_data(_held_buffer).get()
+    else:
+        consumed = True
```

Comment:
I don't know if this should be true when there isn't a held buffer, as a follow on effect would be increasing the duration incorrectly?

Reply:
I set it as true, because we take the current buffer and put it on hold, which (if my understanding of frame consumption is right) means we did consume the frames — even if we'll push it to Mopidy later. 

On the other hand, I'm not sure if the `buffer_timestamp` is increased correctly if its done before emitting data.

Reply:
It does seem weird to attach what we know is the *wrong* timestamp to the buffer we are emitting to GST (although the error will be very small). Should we be doing something more like `buffer_timestamp.increase(_held_buffer.duration)`?

Reply:
I think this is fine. It doesn't matter when we increment `buffer_timestamp` as long as it's before we use in the next delivery.